### ISSUE TITLE
Inconsistent Black Output

### ISSUE THREAD

**Issue by phr00t** [kijai/ComfyUI-DynamiCrafterWrapper#8]
![image](https://github.com/kijai/ComfyUI-DynamiCrafterWrapper/assets/5983470/60d30717-f00a-406e-a9ee-f12b49811f0e)

I haven't figured out why, but sometimes I'll just get black output from DynamiCrafter. As you can see, the DynamicCrafter did get 3 proper 512x512 images and the parameters all looked OK. It took a long time to process the blackness.

![image](https://github.com/kijai/ComfyUI-DynamiCrafterWrapper/assets/5983470/551731c7-f64d-4e94-8c0c-c9fe494cb134)

That above "RuntimeWarning" on line 1433 (the first one) is related to saving the images, as I suspect they are not being outputted correctly from the DynamiCrafter node (so it has nothing proper to save).

I've noticed this a few times, and I would close the server, reopen it, try a different seed and the problem would go away... but it keeps coming back randomly...

**Comment by kijai** [comment]
Yes this seems to happen with some dimensions, unsure why, could be fp16 VAE issue?

**Comment by phr00t** [comment]
I haven't deviated from 512x512 images and I've been using the same model and VAE, so I'm not sure that is related... :(

**Comment by phr00t** [comment]
Closing out and restarting the server and running the same workflow resolves it, at least sometimes...

**Comment by kijai** [comment]
> I haven't deviated from 512x512 images and I've been using the same model and VAE, so I'm not sure that is related... :(

I mean sometimes using VAE in fp16, which we do here unless you change the model loader dtype, can cause NaNs in tensor and thus black images.

**Comment by phr00t** [comment]
Ahhh so the problem doesn't happen in fp32 at all then? That takes 2x the memory in that mode though, right? Hmm....

**Comment by kijai** [comment]
> Ahhh so the problem doesn't happen in fp32 at all then? That takes 2x the memory in that mode though, right? Hmm....

Maybe, I'm not sure, but I just added option to set the VAE dtype only, at these resolutions that VAE doesn't use that much VRAM on it's own so it's fine to use at fp32. The 'auto' setting uses comfy model management detection to determine it. 

Bf16 seems to work as well for the VAE so it should be selected as default on modern GPUs. For some reason this code doesn't work if the unet is cast to bf16 though.

**Comment by phr00t** [comment]
"auto" for me still gave me black output. I'm loading the model as fp16, if that is fine (not sure if it is related). This is probably my biggest issue right now, as I takes >10 minutes to generate something, and 75% of the time its just black :(

**Comment by kijai** [comment]
> "auto" for me still gave me black output. I'm loading the model as fp16, if that is fine (not sure if it is related). This is probably my biggest issue right now, as I takes >10 minutes to generate something, and 75% of the time its just black :(

Try forcing fp32 vae. If it still happens then it's something else, it only happened to me couple of times so it's hard to debug.

**Comment by phr00t** [comment]
> > "auto" for me still gave me black output. I'm loading the model as fp16, if that is fine (not sure if it is related). This is probably my biggest issue right now, as I takes >10 minutes to generate something, and 75% of the time its just black :(
> 
> Try forcing fp32 vae. If it still happens then it's something else, it only happened to me couple of times so it's hard to debug.

I think it is related to step count! If I run the same exact workflow with a step count of 10, I get black. If I run it at 8, it works! It seems like 11-15 was also giving me black, but back up to 20 steps seems to result in normal expected output.

**Comment by kijai** [comment]
Oh yeah changing steps did clear it up the couple of times it happened... assumed it was about just running it again but that could very well be it.
